File: sicpa-root/sicpa-dto/src/main/java/com/sicpa/demo/dto/UserDto.java
package com.sicpa.demo.dto;

import java.io.Serializable;
import java.util.List;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class UserDto extends BaseDto implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long id;
	@NotEmpty
	@Size(max = 60)
	private String password;
	@NotEmpty
	@Size(max = 20)
	private String username;
	private List<RoleUserDto> roleUsers;
	@NotNull
	private EmployeeDto employee;
}

File: sicpa-web/src/app/models/user-dto.ts
import { GenericDto } from "./generic-dto";
import { EmployeeDto } from "./employee-dto";
import { RoleUserDto } from "./role-user-dto";

export class UserDto extends GenericDto {
  id: number;
  password: string;
  username: string;
  roleUsers: Array<RoleUserDto> = [];
  employee: EmployeeDto = new EmployeeDto();
  roles: string[] = [];

  constructor() {
    super();
  }
}
